src/DataStructures/Tensor/Expressions/ScaleBy.hpp

```diff
@@ -0,0 +1,31 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+/// \file
+/// Defines function scale_by
+
+#pragma once
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+
+/*!
+ * \ingroup Functions
+ * \brief scales a tensor by a scalar
+ *
+ * \returns a tensor of the same type as the input tensor
+ *
+ * \details
+ * Divides the input tensor by the given scalar.
+ */
+template <typename TensorType>
+TensorType scale_by(const TensorType& tensor, const DataVector& scale) {
```

Comment:
I don't really understand what you mean, if it was scaling by a constant scalar then the function would just take a double.